### PR TITLE
Update `interfaces` and `gradients` to work with new operator arithmetic

### PR DESCRIPTION
**Pull request by albi3ro**
[sc-58388] [sc-58386]

### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      test directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ ] Ensure that the test suite passes, by running `make test`.

- [ ] Add a new entry to the `doc/releases/changelog-dev.md` file, summarizing the
      change, and including a link back to the PR.

- [ ] The PennyLane source code conforms to
      [PEP8 standards](https://www.python.org/dev/peps/pep-0008/).
      We check all of our code against [Pylint](https://www.pylint.org/).
      To lint modified files, simply `pip install pylint`, and then
      run `pylint pennylane/path/to/file.py`.

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

**Comment by Qottmann** [comment]
I cant run `tests/interfaces/` locally on this branch, any idea why? Jax version 0.4.24
```
=================================================================================================================== ERRORS ===================================================================================================================
_______________________________________________________________________ ERROR collecting interfaces/default_qubit_2_integration/test_autograd_qnode_default_qubit_2.py _______________________________________________________________________
ImportError while importing test module '/Users/korbinian.kottmann/Xanadu/pennylane/tests/interfaces/default_qubit_2_integration/test_autograd_qnode_default_qubit_2.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../../anaconda3/envs/pennylane311/lib/python3.11/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/interfaces/default_qubit_2_integration/test_autograd_qnode_default_qubit_2.py:26: in <module>
    from tests.param_shift_dev import ParamShiftDerivativesDevice
E   ModuleNotFoundError: No module named 'tests.param_shift_dev'
________________________________________________________________________ ERROR collecting interfaces/default_qubit_2_integration/test_torch_qnode_default_qubit_2.py _________________________________________________________________________
ImportError while importing test module '/Users/korbinian.kottmann/Xanadu/pennylane/tests/interfaces/default_qubit_2_integration/test_torch_qnode_default_qubit_2.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../../anaconda3/envs/pennylane311/lib/python3.11/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/interfaces/default_qubit_2_integration/test_torch_qnode_default_qubit_2.py:23: in <module>
    from tests.param_shift_dev import ParamShiftDerivativesDevice
E   ModuleNotFoundError: No module named 'tests.param_shift_dev'
```

File: pennylane/ops/op_math/sum.py
Comment (Qottmann) [inline pennylane/ops/op_math/sum.py:212]: Was this not differentiable before 😬 ?

**Review by Qottmann — COMMENTED** [review]
Finally ran through `tests/gradient/` tests locally, getting

```
FAILED tests/gradients/parameter_shift/test_parameter_shift.py::TestHamiltonianExpvalGradients::test_torch[default.qubit-False] - RuntimeError: Can't call numpy() on Tensor that requires grad. Use tensor.detach().numpy() instead.
FAILED tests/gradients/parameter_shift/test_parameter_shift.py::TestHamiltonianExpvalGradients::test_torch[default.qubit.torch-False] - RuntimeError: Can't call numpy() on Tensor that requires grad. Use tensor.detach().numpy() instead.
```

I am on latest torch `'2.2.1'`

**Review by Qottmann — APPROVED** [review]
> Gradients pass now, but I get this in autograd
> 
> ```
> FAILED tests/interfaces/test_autograd.py::TestBatchTransformExecution::test_no_batch_transform - Failed: DID NOT RAISE <class 'AssertionError'>
> ```

with [dbec613](https://github.com/PennyLaneAI/pennylane/pull/5356/commits/dbec613aca705753c8074cf3fa93901362c376c8) passing now :+1:

When running `tests/interfaces` I still get 
```
..
______ ERROR collecting interfaces/default_qubit_2_integration/test_autograd_qnode_default_qubit_2.py ______
ImportError while importing test module '/Users/korbinian.kottmann/Xanadu/pennylane/tests/interfaces/default_qubit_2_integration/test_autograd_qnode_default_qubit_2.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../../anaconda3/envs/pennylane311/lib/python3.11/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/interfaces/default_qubit_2_integration/test_autograd_qnode_default_qubit_2.py:26: in <module>
    from tests.param_shift_dev import ParamShiftDerivativesDevice
E   ModuleNotFoundError: No module named 'tests.param_shift_dev'
_______ ERROR collecting interfaces/default_qubit_2_integration/test_torch_qnode_default_qubit_2.py ________
ImportError while importing test module '/Users/korbinian.kottmann/Xanadu/pennylane/tests/interfaces/default_qubit_2_integration/test_torch_qnode_default_qubit_2.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../../anaconda3/envs/pennylane311/lib/python3.11/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/interfaces/default_qubit_2_integration/test_torch_qnode_default_qubit_2.py:23: in <module>
    from tests.param_shift_dev import ParamShiftDerivativesDevice
E   ModuleNotFoundError: No module named 'tests.param_shift_dev'
..
========================================= short test summary info ==========================================
ERROR tests/interfaces/default_qubit_2_integration/test_autograd_qnode_default_qubit_2.py
ERROR tests/interfaces/default_qubit_2_integration/test_torch_qnode_default_qubit_2.py
========== 10336 passed, 4595 skipped, 576 xfailed, 1933 warnings, 2 errors in 1889.03s (0:31:29) ==========
```

Same problem as above. I tried re-installing pennylane, start the tests from `pennylane/` so not sure what the problem is 🤔  Since this likely due to my local environment, I think its okay to approve with a second reviewer confirming those pass.

**Review by lillian542 — COMMENTED** [review]
Looks good to me, waiting for tests to finish running locally but will approve once they've all passed

**Pull request merged**
